File: Web.Admin/src/api/services/Center/clientUser/models/QueryClientUserDetailOutput.ts
import type { ClientUserTypeEnum } from "@/api/enums/ClientUserTypeEnum";
import type { GenderEnum } from "@/api/enums/GenderEnum";

/**
 * 获取客户端用户详情输出
 */
export interface QueryClientUserDetailOutput {
	/**
	 * 客户端用户Id
	 */
	userId?: number;
	/**
	 * 
	 */
	userType?: ClientUserTypeEnum;
	/**
	 * 唯一用户标识
	 */
	openId?: string;
	/**
	 * 统一用户标识
	 */
	unionId?: string;
	/**
	 * 手机
	 */
	mobile?: string;
	/**
	 * 昵称
	 */
	nickName?: string;
	/**
	 * 头像
	 */
	avatar?: string;
	/**
	 * 
	 */
	sex?: GenderEnum;
	/**
	 * 最后登录设备
	 */
	lastLoginDevice?: string;
	/**
	 * 最后登录操作系统（版本）
	 */
	lastLoginOS?: string;
	/**
	 * 最后登录浏览器（版本）
	 */
	lastLoginBrowser?: string;
	/**
	 * 最后登录省份
	 */
	lastLoginProvince?: string;
	/**
	 * 最后登录城市
	 */
	lastLoginCity?: string;
	/**
	 * 最后登录Ip
	 */
	lastLoginIp?: string;
	/**
	 * 最后登录时间
	 */
	lastLoginTime?: string;
	/**
	 * 创建时间
	 */
	createdTime?: string;
	/**
	 * 更新时间
	 */
	updatedTime?: string;
	/**
	 * 手机号更新时间
	 */
	mobileUpdateTime?: string;
	/**
	 * 允许修改手机号
	 */
	readonly allowModifyMobile?: boolean;
	/**
	 * 
	 */
	rowVersion?: number;
}

File: Api.Server/src/Center.Service/ClientUser/Dto/QueryClientUserDetailOutput.cs
// Copyright © 2018-Present 小方
// SPDX-License-Identifier: Apache-2.0
// 
// 本文件依据 Apache License 2.0 授权，完整条款见仓库根目录 LICENSE。
// 本软件按“原样”提供，相关免责声明及责任限制以许可证及适用法律为准。
// 版权来源、合法使用与二次开发责任说明见仓库根目录 README.md。

using Fast.Center.Domain;

namespace Fast.Center.Service.ClientUser.Dto;

/// <summary>
/// 获取客户端用户详情输出
/// </summary>
public class QueryClientUserDetailOutput : UpdateVersionInput
{
    /// <summary>
    /// 客户端用户Id
    /// </summary>
    public long UserId { get; set; }

    /// <summary>
    /// 用户类型
    /// </summary>
    public ClientUserTypeEnum UserType { get; set; }

    /// <summary>
    /// 手机
    /// </summary>
    public string Mobile { get; set; }

    /// <summary>
    /// 唯一用户标识
    /// </summary>
    public string OpenId { get; set; }

    /// <summary>
    /// 统一用户标识
    /// </summary>
    public string UnionId { get; set; }

    /// <summary>
    /// 是否已设置密码
    /// </summary>
    public bool HasPassword { get; set; }

    /// <summary>
    /// 昵称
    /// </summary>
    public string NickName { get; set; }

    /// <summary>
    /// 头像
    /// </summary>
    public string Avatar { get; set; }

    /// <summary>
    /// 性别
    /// </summary>
    public GenderEnum Sex { get; set; }

    /// <summary>
    /// 最后登录设备
    /// </summary>
    public string LastLoginDevice { get; set; }

    /// <summary>
    /// 最后登录操作系统（版本）
    /// </summary>
    public string LastLoginOS { get; set; }

    /// <summary>
    /// 最后登录浏览器（版本）
    /// </summary>
    public string LastLoginBrowser { get; set; }

    /// <summary>
    /// 最后登录省份
    /// </summary>
    public string LastLoginProvince { get; set; }

    /// <summary>
    /// 最后登录城市
    /// </summary>
    public string LastLoginCity { get; set; }

    /// <summary>
    /// 最后登录Ip
    /// </summary>
    public string LastLoginIp { get; set; }

    /// <summary>
    /// 最后登录时间
    /// </summary>
    public DateTime? LastLoginTime { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime? CreatedTime { get; set; }

    /// <summary>
    /// 更新时间
    /// </summary>
    public DateTime? UpdatedTime { get; set; }

    /// <summary>
    /// 手机号更新时间
    /// </summary>
    public DateTime? MobileUpdateTime { get; set; }

    /// <summary>
    /// 允许修改手机号
    /// </summary>
    public bool AllowModifyMobile => MobileUpdateTime == null || MobileUpdateTime.Value.AddDays(+1) < DateTime.Now;
}
